connect/client.go

```diff
@@ -101,6 +104,35 @@ type restClient struct {
 	client    httpClient
 }
 
+func (rs *restClient) LoadStructFromItem(i interface{}, itemUUID string, vaultUUID string) error {
+	config, err := checkStruct(i)
+	if err != nil {
+		return err
+	}
+	t := config.Type()
+	item := parsedItem{}
+
+	for i := 0; i < t.NumField(); i++ {
+		value := config.Field(i)
+		field := t.Field(i)
+
+		if !value.CanSet() {
+			return fmt.Errorf("cannot load config into private fields")
+		}
+
+		item.vaultUUID = vaultUUID
+		item.itemUUID = itemUUID
+		item.fields = append(item.fields, &field)
+		item.values = append(item.values, &value)
+	}
+
+	if err := setValuesForTag(rs, &item, false); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Most code here seems to be duplicated between this and `LoadStructFromItemByTitle`. If we now change only one of the two implementations, the functionality drifts.

What happens if you take line 123 and 124 out of the loop? Does that make it possible to move some code to a shared function?

Reply:
Nice one! I've just extracted the common functionality of the two functions, could you please have another look over it?😄

Reply:
Yes, nice job!